### PR TITLE
Add JDK-21 to GA worklflows

### PR DESCRIPTION
**Pull request by reta**
### Description
Add JDK-21 to GA worklflows
 
### Issues Resolved
Part of https://github.com/opensearch-project/OpenSearch/issues/10334
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vamsi-amazon** [comment]
Merged https://github.com/opensearch-project/sql/pull/2476.
Rerunning failed workflows.

File: .github/workflows/integ-tests-with-security.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        java: [ 11, 17 ]
+        java: [ 11, 17, 21 ]
```
Comment (vamsi-amazon) [inline .github/workflows/integ-tests-with-security.yml:23]: @reta  are we going to start supporting 21 in next 2.x version? Lets suppose if 3.0 release is near by can we skip supporting 2.x? How do we make these decisions?

**Review by vamsi-amazon — DISMISSED** [review]

File: .github/workflows/integ-tests-with-security.yml (hunk 1 above)
Comment (reta) [reply .github/workflows/integ-tests-with-security.yml:23]: > @reta are we going to start supporting 21 in next 2.x version? Lets suppose if 3.0 release is near by can we skip supporting 2.x? How do we make these decisions?

@vamsi-amazon 2.12.0 is going to be bundled with JDK-21, see please https://github.com/opensearch-project/OpenSearch/issues/10334, https://github.com/opensearch-project/OpenSearch/pull/11003. The context for it - JDK-21 is the next LTS and there is new vector APIs that Apache Lucene and OpenSearch benefit from, including plugins. Thank you.

Comment (reta) [reply .github/workflows/integ-tests-with-security.yml:23]: > Lets suppose if 3.0 release is near by can we skip supporting 2.x?
I think when 3.0 is out, than 2.x would go to a maintenance mode, as 1.3.x is these days (but fairly speaking, we are just kicking off the discussions to understand when 3.0 could be out). Thank you.

**Review by vamsi-amazon — APPROVED** [review]

**Review by penghuo — APPROVED** [review]

**Review by vamsi-amazon — APPROVED** [review]

**Review by penghuo — APPROVED** [review]

**Pull request merged**